### ISSUE TITLE
Add random function

### ISSUE THREAD

**Issue by josevalim** [whatyouhide/stream_data#15]
It gives us a random element from the `StreamData`. Or it should be part of the `StreamData` module, or we should include a special implementation of it in `Enum.random/2` when we merge this into Elixir, or we should add def random to the `Enumerable` protocol (since a couple types already specialize it).

**Comment by whatyouhide** [comment]
@josevalim can we start with `StreamData.random/1` (`pick/1`?), or should we discuss the inclusion of this in `Enumerable` first?

**Comment by whatyouhide** [comment]
@josevalim we can start with `ExUnitProperties.random/1` that reads the seed from ExUnit configuration?

**Comment by josevalim** [comment]
@whatyouhide we need a random per test otherwise every test will emit the same value because they will have the same seed. So I think we need to first add this feature to ExUnit.

**Comment by josevalim** [comment]
Btw, we should call this `pick`.

**Comment by josevalim** [comment]
And the function should probably be on ExUnit.PropertyTest since it requires using ExUnit's seed.

**Comment by whatyouhide** [comment]
@josevalim for this, should we expose `StreamData.call(data, seed, size)` then? Maybe make it `@doc false`? Also about the generation size, should we use like a random generation size between `1..100`?

**Comment by josevalim** [comment]
@whatyouhide we could also have `StreamData.pick(data, seed, size)` although you may be worried about import conflicts? What do you think? If we are going to have `call` with `@doc false` then it should probably be `__call__`?

**Comment by whatyouhide** [comment]
Yes I meant for it to be `__call__`. And yes we'd have import conflicts so I'd go with that instead of `StreamData.pick/3`. Thoughts?

**Comment by josevalim** [comment]
@whatyouhide sounds good to me for now. :) We can always revisit it later. Can you also sketch how the take_random implementation would look like for the Enumerable protocol? Would we use __call__ too?

**Comment by whatyouhide** [comment]
@josevalim yeah we'd use `__call__` in the `Enumerable` protocol as well.

```elixir
defimpl Enumerable, for: StreamData do
  def take_random(data) do
    seed = :rand.seed_s(:exs1024), # or taken from pdict, whatever work
    {size, _} = :rand.uniform_s(100, seed)
    StreamData.__call__(data, seed, size)
  end
end
```

Something like this?

**Comment by josevalim** [comment]
@whatyouhide awesome. It feels like we should expose it somehow but if we are not comfortable with it yet then `__call__` it is.

**Comment by whatyouhide** [comment]
@josevalim I would wait because I don't want to expose the random seed and the details this easily yet.

**Comment by whatyouhide** [comment]
Closed by #52.
